web/src/site-admin/SiteAdminAddExternalServicesPage.tsx

```diff
@@ -17,45 +18,93 @@ interface Props extends ThemeProps {
 /**
  * Page for choosing a service kind and variant to add, among the available options.
  */
-export class SiteAdminAddExternalServicesPage extends React.Component<Props> {
-    /**
-     * Returns the id of the external service from the URL parameters.
-     */
-    private getExternalServiceID(): string | null {
-        return new URLSearchParams(this.props.history.location.search).get('id') ?? null
+export const SiteAdminAddExternalServicesPage: React.FunctionComponent<Props> = props => {
+    const [hasDismissedPrivacyWarning, setHasDismissedPrivacyWarning] = useLocalStorage(
+        'HAS_DISMISSED_CODEHOST_PRIVACY_WARNING',
```

Comment:
Nit: usually use kebab-case or camelCase
```suggestion
        'hasDismissedCodeHostPrivacyWarning',
```